Source/missiles.cpp

```diff
@@ -2284,14 +2284,26 @@ void AddFirewall(int mi, int sx, int sy, int dx, int dy, int midir, char mienemy
 {
 	int i;
 
-	missile[mi]._midam = (random_(53, 10) + random_(53, 10) + plr[id]._pLevel + 2) << 4;
+#ifdef HELLFIRE
+	missile[mi]._midam = (random_(53, 10) + random_(53, 10) + 2 + (id > 0) ? plr[id]._pLevel : currlevel);

```

Comment:
This was probably supposed to be:
```CPP
	missile[mi]._midam = (random_(53, 10) + random_(53, 10) + 2 + ((id > 0) ? plr[id]._pLevel : currlevel));
```
Do we know if firewalls are horribly bugged in hellfire?

Reply:
well, it wouldn't bug them "horribly" but cause firewall traps to get dmg from player level and not currlevel, right?

Reply:
It also strips all the random from the damage, it effectively becomes:
`	missile[mi]._midam = plr[id]._pLevel;`
Where for diablo it was:
`	missile[mi]._midam = random_(53, 10) + random_(53, 10) + 2 + plr[id]._pLevel;`

Reply:
right lol xD